app/controllers/providers/allocations_controller.rb

```diff
@@ -5,6 +5,7 @@ class AllocationsController < ApplicationController
     before_action :build_training_provider, except: %i[index initial_request]
     before_action :require_provider_to_be_accredited_body!
     before_action :require_admin_permissions!
+    before_action :build_allocation, only: %i[update edit_repeat_request]
```

Comment:
We may want to consider refactoring these before action callbacks
See https://ukgovernmentdfe.slack.com/archives/CGVNJ2Q2F/p1589301415297900?thread_ts=1589291822.294000&cid=CGVNJ2Q2F

Reply:
I agree with @asmega in the Slack chat about getting rid of this. This one is also incorrectly named as we're not building it, we're retrieving it from the API. That's an important distinction here because it means we have the saved state of the `Allocation` so we don't really need a separate edit action (edit_repeat_request). We can just edit and render a different view (or view in a different state) depending on the state of the retrieved allocation.

We can probably have a wider discussion about the slack thread but this defo makes more sense as:

```
def allocation
  @_allocation ||= Allocation.find(<whatevs>)
end
```